UnitTests/BraintreePayPalTests/BTPayPalClient_Tests.swift

```diff
@@ -670,6 +689,41 @@ class BTPayPalClient_Tests: XCTestCase {
         BTPayPalClient.handleReturnURL(URL(string: "https://mycoolwebsite.com/braintree-payments/success")!)
         XCTAssertNil(BTPayPalClient.payPalClient)
     }
+    
+    // MARK: - App Switch - tokenize
+
+    func testTokenizeVaultAccount_whenPayPalAppApprovalURLPresent_attemptsAppSwitchWithParameters() async {
+        let fakeApplication = FakeApplication()
+        payPalClient.application = fakeApplication
+        
+        mockAPIClient.cannedResponseBody = BTJSON(value: [
+            "paymentResource": [
+                "paypalAppApprovalUrl": "https://www.some-url.com/some-path?token=value1",
+                "redirectUrl": "https://www.other-url.com/"
+            ]
+        ])
+        
+        let vaultRequest = BTPayPalVaultRequest(userAuthenticationEmail: "fake@gmail.com", enablePayPalAppSwitch: true)
+        
+        payPalClient.tokenize(vaultRequest) { _, _ in }
+
+        XCTAssertTrue(fakeApplication.openURLWasCalled)
+        
+        let urlComponents = URLComponents(url: fakeApplication.lastOpenURL!, resolvingAgainstBaseURL: true)
+        XCTAssertEqual(urlComponents?.host, "www.some-url.com")
+        XCTAssertEqual(urlComponents?.path, "/some-path")
+
+        XCTAssertEqual(urlComponents?.queryItems?[0].name, "source")
+        XCTAssertEqual(urlComponents?.queryItems?[0].value, "braintree_sdk")
+        XCTAssertEqual(urlComponents?.queryItems?[1].name, "switch_initiated_time")
+        if let urlTimestamp = urlComponents?.queryItems?[1].value {
+            XCTAssertNotNil(Int(urlTimestamp))
+        } else {
+            XCTFail("Expected integer value for query param `switch_initiated_time`")
+        }
```

Comment:
```suggestion
        XCTAssertNotNil(Int(urlComponents?.queryItems?[1].value ?? ''))
```
🧶 : Take or leave to avoid if statement in the test.